src/main/java/org/motechproject/mots/repository/ModuleRepository.java

```diff
@@ -1,11 +1,19 @@
 package org.motechproject.mots.repository;
 
 import java.util.Optional;
+import java.util.Set;
 import java.util.UUID;
 import org.motechproject.mots.domain.Module;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
 
 public interface ModuleRepository extends CrudRepository<Module, UUID> {
 
   Optional<Module> findById(UUID id);
+
+  @Query("SELECT DISTINCT mp.module FROM ModuleProgress mp "
```

Comment:
you don't need a querry here, if you rename the method to findByCommunityHealthWorkerIdAndStatusNot and add status parameter.
you can remove this method, see the comments below